### PR TITLE
[Improvement-9338][API] show more create datasource exception message

### PR DESCRIPTION
**Pull request by 1qase4**
This pull request fix Execption message miss, modify file dolphinscheduler-api\src\main\java\org\apache\dolphinscheduler\api\service\impl\DataSourceServiceImpl.java

Because of org.apache.dolphinscheduler.plugin.datasource.api.client.CommonDataSourceClient#checkClient method convert sqlException to RuntimeException, org.apache.dolphinscheduler.api.service.impl.DataSourceServiceImpl#checkConnection method can't get right exception message.
image

This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by SbloodyS** [comment]
Hi @1qase4, please create an issue link to this pr. Thanks.

**Comment by 1qase4** [comment]
ok   close this pr first, Thanks.

**Comment by SbloodyS** [comment]
> ok close this pr first, Thanks.

Hi, why did you close this pr?

**Comment by 1qase4** [comment]
After waiting for issue confirmation, i  will reopen this pr again.

**Comment by SbloodyS** [comment]
> After waiting for issue confirmation, i will reopen this pr again.

There is no need for issue confirmation. All you need is link the issue to this pr. eg: describe ```this pr will close #9338``` in this pr. To target specific issue from PR, we recommend use keyword ```fix: #issue_id``` or ```close: #issue_id``` or ```closes: #issue_id``` in your PR describe(not title, jus desc). It would not only connect issue to PR but also close issue automatically when PR is be closed. Feel free to reopen this pr.

BTW:
As your're the first time contributor, your CI could run until committer allow it. Once you want to run CI, you can @ me or other members/committer in ```https://dolphinscheduler.apache.org/en-us/community/team.html``` to approve to run CI.

**Comment by 1qase4** [comment]
close #9338

**Comment by 1qase4** [comment]
@SbloodyS  please run CI once. Thanks.

**Comment by SbloodyS** [comment]
> @SbloodyS please run CI once. Thanks.

I had approved to run it.

**Comment by SbloodyS** [comment]
@caishunfeng PTAL. Thanks.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Review by mithindev — COMMENTED** [review]
Reviewed. Makes the code better.

**Pull request merged**
